tests/php/Service/ThemeResolverTest.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace SilverStripe\Subsites\Tests\Service;
+
+use SilverStripe\Core\Config\Config;
+use SilverStripe\Dev\SapphireTest;
+use SilverStripe\Subsites\Model\Subsite;
+use SilverStripe\Subsites\Service\ThemeResolver;
+use SilverStripe\View\SSViewer;
+
+class ThemeResolverTest extends SapphireTest
+{
+    protected $themeList = [
+        SSViewer::PUBLIC_THEME,
+        'custom',
+        'main',
+        'backup',
+        SSViewer::DEFAULT_THEME,
+    ];
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        // Setup known theme config
+        Config::modify()->set(SSViewer::class, 'themes', $this->themeList);
+    }
+
+    public function testSubsiteWithoutThemeReturnsDefaultThemeList()
+    {
+        $subsite = new Subsite();
+        $resolver = new ThemeResolver();
+
+        $this->assertSame($this->themeList, $resolver->getThemeList($subsite));
+    }
+
+    public function testSubsiteWithCustomThemePrependsToList()
+    {
+        $subsite = new Subsite();
+        $subsite->Theme = 'subsite';
+
+        $resolver = new ThemeResolver();
+
+        $expected = array_merge(['subsite'], $this->themeList);
+
+        $this->assertSame($expected, $resolver->getThemeList($subsite));
+    }
+
+    public function testSubsiteWithCustomThemeDoesNotCascadeUpTheList()
+    {
+        $subsite = new Subsite();
+        $subsite->Theme = 'main';
+
+        $resolver = new ThemeResolver();
+
+        $expected = [
+            'main', // 'main' is moved to the top
+            SSViewer::PUBLIC_THEME, // $public is preserved
```

Comment:
I'm not sure this is a realistic use case - public is put first in theme lists

Reply:
This is how it works currently - the chosen theme is moved _above_ public. It'd technically be breaking if I changed this.